types/types.go

```diff
@@ -1040,6 +1042,23 @@ func CreateDisksFromAnnotation(annotation string) (map[string]longhorn.DiskSpec,
 		if disk.StorageReserved < 0 || disk.StorageReserved > diskStat.StorageMaximum {
 			return nil, fmt.Errorf("the storageReserved setting of disk %v is not valid, should be positive and no more than storageMaximum and storageAvailable", disk.Path)
 		}
+		if disk.StorageReserved == 0 {
+			if disk.Type == longhorn.DiskTypeBlock {
+				file, err := os.Open(ReplicaHostPrefix + disk.Path)
+				if err != nil {
+					return nil, fmt.Errorf("failed to open block device at %s: %w", disk.Path, err)
+				}
+				defer file.Close()
+				var size uint64
+				_, _, errno := unix.Syscall(unix.SYS_IOCTL, file.Fd(), 0x80081272, uintptr(unsafe.Pointer(&size)))
+				if errno != 0 {
+					return nil, fmt.Errorf("failed to get block device size for %s: errno=%v", disk.Path, errno)
+				}
+				disk.StorageReserved = int64(size) * storageReservedPercentage / 100
+			} else {
+				disk.StorageReserved = diskStat.StorageMaximum * storageReservedPercentage / 100
```

Comment:
Good catch. I just found Longhorn hadn't handled the `StorageReserved` for filesystem-type disk as well...
cc @shuo-wu @PhanLe1010 @c3y1huang @innobead